### ISSUE TITLE
OnInit Trigger for Page Extensions

### ISSUE THREAD

**Issue by megel** [Microsoft/AL#1968]
Is it possible to add **OnInit** trigger for a Page-Extensions?

**Comment by StanislawStempin** [comment]
Do you have any requirements around this potential trigger? Should it run after the base page OnInit, in which case this would be OnAfterInit?
In any case we will discuss internally and see how we can accommodate your suggestion.

**Comment by megel** [comment]
My requirement is to use this trigger for:

* initial checks
* initialization
* change the behavior of the page (visibility / user specific rights)
 
I believe `OnOpenPage` will be too late in some cases.

The `OnAfterInit` would be fine.

**Comment by JSheeper** [comment]
Yes, I also think the trigger would be very useful and I don't see any reasons for not providing it.

**Comment by ShadowXVII** [comment]
We have run into this too. OnAfterInit would be fine.

**Comment by dzakojc** [comment]
We also need to add some code in trigger OnInit of the pageextension, so OnAfterInit would be very useful, indeed. Thank you Damjan

**Comment by SBalslev** [comment]
Hi! This issue is stale - sometimes great ideas go stale if not enough people talk about them. Please go ahead and post this to our Ideas forum at https://aka.ms/BusinessCentralideas, or vote up the idea if its already there. We're constantly monitoring top Ideas and will consider them for a future release.
